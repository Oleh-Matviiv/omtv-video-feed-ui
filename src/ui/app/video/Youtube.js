import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Missing from './Missing';

export const Play = styled.iframe.attrs({
  className: 'video-play-youtube',
})`
  grid-area: play;
`;

const Youtube = ({ id }) => id ? (
  <Play
    width="100%"
    height="400px"
    src={ `https://www.youtube.com/embed/${id}` }
  />
) : <Missing />;
Youtube.propTypes = {
  id: PropTypes.string,
};

export default Youtube;
