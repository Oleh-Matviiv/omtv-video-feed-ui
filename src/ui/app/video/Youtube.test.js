import React from 'react';
import { mount, shallow } from 'enzyme';
import 'jest-styled-components';

import Youtube, { Play } from './Youtube';

describe('snapshots', () => {
  it('Youtube', () => {
    const wrapper = shallow(
      <Youtube id="tgbNymZ7vqY" />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Play', () => {
    const wrapper = mount(
      <Play width="200px" src="http://g.com/1.mp4">
        children
      </Play>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
