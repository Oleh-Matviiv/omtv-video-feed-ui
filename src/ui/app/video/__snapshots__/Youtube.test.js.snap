// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Play 1`] = `
.c0 {
  grid-area: play;
}

<styled.iframe
  src="http://g.com/1.mp4"
  width="200px"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "video-play-youtube",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  grid-area: play;
",
          ],
        },
        "displayName": "styled.iframe",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "iframe",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    src="http://g.com/1.mp4"
    width="200px"
  >
    <iframe
      className="video-play-youtube c0"
      src="http://g.com/1.mp4"
      width="200px"
    >
      children
    </iframe>
  </StyledComponent>
</styled.iframe>
`;

exports[`snapshots Youtube 1`] = `
<styled.iframe
  height="400px"
  src="https://www.youtube.com/embed/tgbNymZ7vqY"
  width="100%"
/>
`;
